api/tests/core/bookings/test_api.py

```diff
@@ -737,6 +738,46 @@ def test_cancel_booking(self, mocked_cancel_external_booking):
             TransactionalEmail.BOOKING_CANCELLATION_BY_BENEFICIARY_TO_PRO.value
         )  # to offerer
 
+    @patch("pcapi.tasks.external_api_booking_notification_tasks.external_api_booking_notification_task.delay")
+    def test_send_notification_to_external_api_when_cancel_booking(self, mocked_task):
+        provider = providers_factories.ProviderFactory(
+            name="Technical provider", localClass=None, notificationExternalUrl="http://external.api.com/notify"
+        )
+        providers_factories.OffererProviderFactory(provider=provider)
+        stock = offers_factories.EventStockFactory(
+            offer__lastProvider=provider,
+            offer__venue__address="1 boulevard Poissonniere",
+            offer__extraData={"ean": "1234567890123"},
+            offer__withdrawalType=offers_models.WithdrawalTypeEnum.NO_TICKET,
+            dnBookedQuantity=1,
+            idAtProviders="",
+            lastProvider=provider,
+        )
+        booking = bookings_factories.BookingFactory(stock__dnBookedQuantity=1, stock=stock)
```

Comment:
Tu as déjà défini `dnBookedQuantity` juste au-dessus, pas besoin de le passer à nouveau en argument.
```suggestion
        booking = bookings_factories.BookingFactory(stock=stock)
```

.. et d'ailleurs, je me demande à quoi sert de définir `dnBookedQuantity` dans le `EventStockFactory`.

Reply:
ça sert pour pouvoir valider que la quantité de stock est bien mise à jour après une réservation ou une annulation

Reply:
Comme l'a fait remarquer Dan, on ne fait aucune validation là-dessus dans le test que tu as ajouté. Et c'est une bonne chose : ça n'est pas l'objet de ce test.